src/Sulu/Bundle/ContentBundle/Search/Metadata/StructureProvider.php

```diff
@@ -130,13 +130,17 @@ public function getMetadata(Metadata $documentMetadata, StructureMetadata $struc
                             continue;
                         }
 
+                        $tag = $componentProperty->getTag('sulu.search.field');
+                        $tagAttributes = $tag['attributes'];
+
                         $propertyMapping->addFieldMapping(
-                            'title',
+                            $property->getName() . '.' . $componentProperty->getName(),
                             array(
-                                'type' => 'string',
+                                'type' => isset($tagAttributes['type']) ? $tagAttributes['type'] : 'string',
                                 'field' => $this->factory->createMetadataProperty(
                                     '[' . $componentProperty->getName() . ']'
-                                )
+                                ),
+                                'indexed' => !isset($tagAttributes['index']) || $tagAttributes['index'] !== 'false',
```

Comment:
Now I slowly can see what this all is about... But I am still not sure if this right, since the `'indexed'` key seems to be `false` always...